porch/engine/pkg/kpt/render_test.go

```diff
@@ -0,0 +1,192 @@
+func TestRenderWithFunctionConfigFile(t *testing.T) {
+	t.Skip("kpt renderer does not correctly construct function config")
+
+	fs := &memfs{}
+	if err := fs.MkdirAll("/simple-bucket"); err != nil {
+		t.Errorf("Failed MkdirAll: %v", err)
+	}
+	for k, v := range map[string]string{
+		"/simple-bucket/bucket.yaml":  simpleBucketBucket,
+		"/simple-bucket/Kptfile":      simpleBucketKptfile,
+		"/simple-bucket/setters.yaml": simpleBucketSetters,
+	} {
+		if err := fs.WriteFile(k, []byte(v)); err != nil {
+			t.Errorf("Failed creating file %q: %v", k, err)
+		}
+	}
+
+	r := render.Renderer{
+		PkgPath:    "/simple-bucket",
+		Runner:     &runner{},
+		FileSystem: fs,
+	}
+
+	if err := r.Execute(fake.CtxWithDefaultPrinter()); err != nil {
+		t.Errorf("Render failed: %v", err)
+	}
+
+	got, err := fs.ReadFile("/simple-bucket/bucket.yaml")
+	if err != nil {
+		t.Errorf("Cannot read \"/simple-bucket/bucket.yaml\": %v", err)
+	}
+
+	if diff := cmp.Diff(wantBucketBucket, string(got)); diff != "" {
+		t.Errorf("Unexpected result (-want, +got): %s", diff)
+	}
+}
+
+const (
+	inlineBucketBucket = `
+apiVersion: storage.cnrm.cloud.google.com/v1beta1
+kind: StorageBucket
+metadata: # kpt-merge: config-control/blueprints-project-bucket
+  name: blueprints-project-bucket # kpt-set: ${project-id}-${name}
+  namespace: config-control # kpt-set: ${namespace}
+  annotations:
+    cnrm.cloud.google.com/force-destroy: "false"
+    cnrm.cloud.google.com/project-id: blueprints-project # kpt-set: ${project-id}
+spec:
+  storageClass: standard # kpt-set: ${storage-class}
+  uniformBucketLevelAccess: true
+  versioning:
+    enabled: false
+`
+
+	inlineBucketKptfile = `
+apiVersion: kpt.dev/v1
+kind: Kptfile
+metadata:
+  name: simple-bucket
+  annotations:
+    blueprints.cloud.google.com/title: Google Cloud Storage Bucket blueprint
+info:
+  description: A Google Cloud Storage bucket
+pipeline:
+  mutators:
+    - image: gcr.io/kpt-fn/apply-setters:v0.2.0
+      configMap:
+        name: updated-bucket-name
+        namespace: updated-namespace
+        project-id: updated-project-id
+        storage-class: updated-storage-class
+`
+
+	wantBucketBucket = `apiVersion: storage.cnrm.cloud.google.com/v1beta1
+kind: StorageBucket
+metadata: # kpt-merge: config-control/blueprints-project-bucket
+  name: updated-project-id-updated-bucket-name # kpt-set: ${project-id}-${name}
+  namespace: updated-namespace # kpt-set: ${namespace}
+  annotations:
+    cnrm.cloud.google.com/force-destroy: "false"
+    cnrm.cloud.google.com/project-id: updated-project-id # kpt-set: ${project-id}
+    cnrm.cloud.google.com/blueprint: 'kpt-fn'
+spec:
+  storageClass: updated-storage-class # kpt-set: ${storage-class}
+  uniformBucketLevelAccess: true
+  versioning:
+    enabled: false
+`
+)
+
+func TestRenderWithFunctionConfigInline(t *testing.T) {
+	fs := &memfs{}
+	if err := fs.MkdirAll("/inline-bucket"); err != nil {
+		t.Errorf("Failed MkdirAll: %v", err)
+	}
+	for k, v := range map[string]string{
```

Comment:
Not necessarily for this PR, but a pattern I like is to have a directory with inputs under testdata for each test, and simply read the directory and in this case load it into `fs`